awx_collection/tools/templates/tower_module.j2

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+{# The following is set by the generate.yml file:
+ #   item_type: the type of item i.e. 'teams'
+ #   human_readable: the type with _ replaced with spaces i.e. worflow job template
+ #   singular_item_type: the type of an item replace singularized i.e. team
+ #   type_map: a mapping of things like string to str
+ #}
+{% set name_option = 'username' if item_type == 'users' else 'name' %}
+
+# (c) {{ ansible_date_time['year'] }}, John Westcott IV <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_{{ singular_item_type }}
+author: "John Westcott IV (@john-westcott-iv)"
```

Comment:
Might be nice to include something in the author details about it being auto-generated.